resources/planwise/plpgsql/isochrones.sql

```diff
@@ -79,22 +82,29 @@ begin
   facility_geom := (select the_geom from facilities where id = f_id);
   closest_node_geom := (select the_geom from ways_vertices_pgr where id = facility_node);
 
-  IF NOT EXISTS (SELECT r.id FROM regions AS r WHERE ST_Contains(r.the_geom, facility_geom) LIMIT 1) THEN
+  country := (SELECT r.country FROM REGIONS AS r WHERE ST_Contains(r.the_geom, facility_geom) AND r.admin_level = 2 LIMIT 1);
+
+  IF country IS NULL THEN
     UPDATE facilities SET processing_status = 'outside-regions' WHERE id = f_id;
     RAISE NOTICE 'warning: Facility % not processed, it is outside the regions boundaries.', f_id;
-    RETURN 'outside-regions';
+
+    SELECT 'outside-regions'::TEXT, NULL::TEXT into ret; RETURN ret;
   END IF;
 
   IF ST_Distance(ST_GeogFromWKB(facility_geom), ST_GeogFromWKB(closest_node_geom)) > 1000 THEN
     UPDATE facilities SET processing_status = 'no-road-network' WHERE id = f_id;
     RAISE NOTICE 'warning: Facility % not processed, it is too far from the road network.', f_id;
-    RETURN 'no-road-network';
+    SELECT 'no-road-network'::TEXT, NULL::TEXT into ret; RETURN ret;
   END IF;
 
+  -- apply an upper bound to the reachable region to avoid retrieving all ways
+  -- bound = area that can be covered travelling in a straight line at 85 km/h for the maximum threshold time
+  bounding_radius_meters := (threshold_finish / 60.0) * 85 * 1000;
+
   insert into edges_agg_cost (
     select e.edge, e.agg_cost, e.node
     from pgr_drivingdistance(
-      'select gid as id, source, target, cost_s as cost from ways',
+      'select gid as id, source, target, cost_s as cost from ways where the_geom @ (select ST_Buffer(the_geom::geography, '|| bounding_radius_meters || ')::geometry from facilities where id = ' || f_id || ' limit 1)',
```

Comment:
You already have the `facility_geom`. Isn't it possible to use it to calculate the bounding area instead of making another subquery? It should be shorter and simpler code.